go/arrow/csv/reader.go

```diff
@@ -469,6 +469,10 @@ func (r *Reader) initFieldConverter(bldr array.Builder) func(string) {
 		return func(s string) {
 			r.parseList(bldr, s)
 		}
+	case arrow.BinaryDataType:
+		return func(s string) {
+			r.parseBinaryDataType(bldr, s)
+		}
```

Comment:
Can you add a case for `*arrow.LargeStringType` above this (such as with the string case) so that it doesn't get dropped into this case too? (I'm pretty sure the LargeString type does meet this interface)